hera_pspec/pspecdata.py

```diff
@@ -2123,6 +2251,8 @@ def pspec(self, bls1, bls2, dsets, pols, n_dlys=None,
                 pol_ints = []
                 pol_cov=[]
 
+                self.set_pol(p[0]) # used in get_Q function to specify the correct polarization for the beam
```

Comment:
If two different polarizations, along with beam are provided, it would result in an error message. If the beam is not provided, it seems to work fine.

Reply:
This is now being done differently, using `feed_pol` variable. Only one polarization is used when beam is provided, else an isotropic beam is assumed (with a warning of unnormalized power spectrum).